axelrod/match_generator.py

```diff
@@ -56,39 +60,48 @@ class RoundRobinMatches(MatchGenerator):
 
     clone_opponents = True
 
-    def build_matches(self, noise=0):
+    def build_match_chunks(self, noise=0):
         """
         A generator that returns player index pairs and match objects for a
         round robin tournament.
 
         Parameters
         ----------
-        noise : float
+        noise : float, 0
             The probability that a player's intended action should be flipped
 
         Yields
         -------
-        tuple
-            player pair index, match object
+        tuples
+            ((player1 index, player2 index), match object)
         """
         for player1_index in range(len(self.players)):
             for player2_index in range(player1_index, len(self.players)):
-                pair = (
-                    self.players[player1_index], self.opponents[player2_index])
-                match = self.build_single_match(pair, noise)
-                yield (player1_index, player2_index), match
+                match_params = self.build_single_match_params(noise)
+                index_pair = (player1_index, player2_index)
+                yield (index_pair, match_params, self.repetitions)
 
-    def build_single_match(self, pair, noise=0):
+    def build_single_match_params(self, noise=0):
         """Create a single match for a given pair"""
-        return Match(
-            pair, self.turns, self.game, self.deterministic_cache, noise)
+        return (self.turns, self.game, None, noise)
```

Comment:
It's for the deterministic cache, which I'm not passing in any more (for this PR at least). Perhaps a comment is ok for now, or a keyword argument? It's possible that someone would want to do so... otherwise we could reorder the parameters of `Match` but I didn't want to do that.